index/scorch/snapshot_index.go

```diff
@@ -190,21 +190,58 @@ func (i *IndexSnapshot) newIndexSnapshotFieldDict(field string,
 
 func (i *IndexSnapshot) FieldDict(field string) (index.FieldDict, error) {
 	return i.newIndexSnapshotFieldDict(field, func(i segment.TermDictionary) segment.DictionaryIterator {
-		return i.Iterator()
+		return i.AutomatonIterator(nil, nil, nil)
 	}, false)
 }
 
+func exclusiveEndFromInclusiveEnd(inclusiveEnd []byte) []byte {
+	rv := inclusiveEnd
+	if len(inclusiveEnd) > 0 {
+		rv = make([]byte, len(inclusiveEnd))
+		copy(rv, inclusiveEnd)
+		if rv[len(rv)-1] < 0xff {
+			// last byte can be incremented by one
+			rv[len(rv)-1]++
+		} else {
+			// last byte is already 0xff, so append 0
+			// next key is simply one byte longer
+			rv = append(rv, 0x0)
+		}
+	}
+	return rv
+}
+
 func (i *IndexSnapshot) FieldDictRange(field string, startTerm []byte,
 	endTerm []byte) (index.FieldDict, error) {
 	return i.newIndexSnapshotFieldDict(field, func(i segment.TermDictionary) segment.DictionaryIterator {
-		return i.RangeIterator(string(startTerm), string(endTerm))
+		endTermExclusive := exclusiveEndFromInclusiveEnd(endTerm)
+		return i.AutomatonIterator(nil, startTerm, endTermExclusive)
 	}, false)
 }
 
+// incrementBytesPrefix produces the first key that does not
```

Comment:
incrementBytesPrefix should be exclusiveEndFromPrefix here, although you're incrementing bytes on the prefix :)

Possibly a name change - maybe obtainExclusiveEndKeyForPrefix ?

Reply:
i went with calculate, not obtain, as that felt more correct to me